llpc/context/llpcCompiler.cpp

```diff
@@ -2271,6 +2277,46 @@ void helperThreadBuildRayTracingPipelineElf(IHelperThreadProvider *helperThreadP
   helperThreadPayload->compiler->releaseContext(context);
 }
 
+// =====================================================================================================================
+// Limited implementation of Llpc::IHelperThreadProvider to support -add-rt-helpers.
+//
+// If no deferred work helper thread providers is available when additional threads are requested via -add-rt-helpers
+// then use an instances of this class to coordinate helper threads.
+class InternalHelperThreadProvider : public Llpc::IHelperThreadProvider {
+public:
+  InternalHelperThreadProvider() {}
+  ~InternalHelperThreadProvider() {}
+
+  virtual void SetTasks(ThreadFunction *pFunction, uint32_t numTasks, void *pPayload) override {
```

Comment:
Not related to your change: How does the interface of `IHelperThreadProvider` allow threads to access the payload passed here?

Reply:
LlpcHelperThreadProvider has access to a DeferredWorkload structure specified by Vulkan, and this is provided by and accessible to the application program.